test/test_network.py

```diff
@@ -48,33 +51,42 @@ def test_send(self):
         self.assertEqual(msg.arbitration_id, 0x12345)
         self.assertTrue(msg.is_extended_id)
 
-        bus.shutdown()
-        self.network.disconnect()
+    def test_send_periodic(self):
+        from threading import Event
 
-    def test_send_perodic(self):
-        bus = can.interface.Bus(interface="virtual", channel=1)
-        self.network.connect(interface="virtual", channel=1)
+        DATA1 = bytes([1, 2, 3])
+        DATA2 = bytes([4, 5, 6])
+        COB_ID = 0x123
+        TIMEOUT = 0.1
+        self.network.connect(
+            interface="virtual",
+            channel=1,
+            receive_own_messages=True
+        )
+        self.addCleanup(self.network.disconnect)
 
-        task = self.network.send_periodic(0x123, [1, 2, 3], 0.01)
-        time.sleep(0.1)
-        # FIXME: This test is a little fragile, as the number of elements
-        #        depends on the timing of the machine.
-        print(f"Queue size: {bus.queue.qsize()}")
-        self.assertTrue(9 <= bus.queue.qsize() <= 13)
-        msg = bus.recv(0)
-        self.assertIsNotNone(msg)
-        self.assertSequenceEqual(msg.data, [1, 2, 3])
-        # Update data
-        task.update([4, 5, 6])
-        time.sleep(0.02)
-        while msg is not None and msg.data == b'\x01\x02\x03':
-            msg = bus.recv(0)
-        self.assertIsNotNone(msg)
-        self.assertSequenceEqual(msg.data, [4, 5, 6])
-        task.stop()
+        acc = []
+        event = Event()
+
+        def hook(id_, data, ts):
+            acc.append(data)
```

Comment:
We *could* also add an accumulator for the timestamps and do an `assertAlmostEqual` to validate the interval.

Reply:
Sounds good, although a longer interval will probably help to get better accuracy here.

Reply:
Resolved with 0af94f7ed1b35266ac9a6b1344467f96f09ab4c6.